model/process.go

```diff
@@ -1,17 +1,35 @@
 package model
 
 import (
+	"errors"
+
 	"github.com/elastic/apm-server/utility"
 	"github.com/elastic/beats/libbeat/common"
 )
 
 type Process struct {
-	Pid   int
+	Pid   *int
 	Ppid  *int
 	Title *string
 	Argv  []string
 }
 
+func (p *Process) Decode(input interface{}) error {
+	raw, ok := input.(map[string]interface{})
+	if raw == nil {
+		return nil
+	}
+	if !ok {
+		return errors.New("Invalid type for process")
```

Comment:
i think this line can not be ever executed, if type assert fails `raw` will be `nil` and function will return 2 lines above.

Reply:
found and fixed that, it should be `if input == nil..`